---
format: 
  html:
    toc: true
execute:
  echo: true
---

# OLS Analysis of Heat Stress: Socioeconomic and Built Form Relationships

**1. Analyzing Heat Stress Around Bus Stops Using OLS**

Using Ordinary Least Squares (OLS) to analyze the relationship between heat stress around bus stops and socioeconomic and built form factors. We adopt a 400-meter buffer area for each bus stop, establishing a standardized buffer that effectively captures the urban context surrounding each stop. This distance, widely recognized in transit research as a reasonable walking limit for bus commuters, allows for a meaningful representation of each stop’s immediate environment. Within this buffer, we calculate the weighted average values of all socioeconomic variables.



In [2]:
import pickle
import pandas as pd

# File paths
model_file_path = "ols_model.pkl"
vif_file_path = "vif_results.csv"

# Load the OLS model
with open(model_file_path, "rb") as file:
    loaded_model = pickle.load(file)

# Load the VIF data
loaded_vif_data = pd.read_csv(vif_file_path)

# Print results
print(loaded_model.summary())
print("\nLoaded VIF Results:")
print(loaded_vif_data)


                            OLS Regression Results                            
Dep. Variable:                   MEAN   R-squared:                       0.715
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     1355.
Date:                Wed, 25 Dec 2024   Prob (F-statistic):               0.00
Time:                        04:15:22   Log-Likelihood:                -1190.0
No. Observations:                8126   AIC:                             2412.
Df Residuals:                    8110   BIC:                             2524.
Df Model:                          15                                         
Covariance Type:            nonrobust                                         
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                   37.6897 

**2. Discussion of the results**

- ***Model Fit***:
The results indicate that the dependent variable mean temperature of bus stops is well-explained by the model, with an R-squared value of 0.715 and an adjusted R-squared of 0.714, meaning approximately 71.4% of the variation in mean temperature of bus stops is captured by the included independent variables. 

- ***Key Predictors***:
The regression model reveals that socioeconomic, environmental, and built-environment factors significantly influence the mean temperature at bus stops. Socioeconomic factors such as the proportion of Native Americans show a strong positive relationship with bus stop temperatures (coefficient 3.4687), while higher proportions of Asian Americans and other racial groups are associated with lower temperatures. Higher poverty rates also correlate with elevated bus stop temperatures, reflecting disparities in environmental conditions.
Environmental factors like the GINI Index (income inequality) and population density have smaller impacts, with higher population density contributing to slightly higher temperatures, likely due to urban heat island effects. Green space, represented by the Green View Index (GVI), has a strong cooling effect, with a significant negative coefficient (-6.8951), indicating that greenery plays a key role in reducing temperatures at bus stops.
Built-environment features such as Floor Area Ratio (FAR) and enclosure also influence temperatures. Higher FAR (denser areas) slightly reduces temperatures, while enclosed areas significantly mitigate heat. These findings highlight the importance of urban design interventions, including increasing greenery and improving shading, to address elevated bus stop temperatures in densely populated or disadvantaged areas.

- ***VIF***:
The VIF values indicate that the model does not suffer from severe multicollinearity,